src/hotspot/share/runtime/os.cpp

```diff
@@ -1857,3 +1845,82 @@ void os::naked_sleep(jlong millis) {
   }
   naked_short_sleep(millis);
 }
+
+
+////// Implementation of pagesizeset_t
+
+// A pagesizeset_t is a set containing a set of page sizes.
+
+// sets the given page size
+void os::PagesizeSet::add(size_t pagesize) {
+  assert(is_power_of_2(pagesize), "pagesize must be a power of 2: " INTPTR_FORMAT, pagesize);
+  _v |= pagesize;
+}
+
+// returns true if given page size is part of the set
```

Comment:
Not possible, since its used from within os::Linux, os::Aix etc.

Reply:
Your comment is probably related to the comment of making the `PageSizeSet` class private. Hmm, I did try that out though and compiled locally on Linux.... maybe I messed up somewhere. Leave it public then.

Otherwise I am a bit confused how this statement relates to the request to move the code comment to the .hpp file.

Reply:
You know, communication was a bit easier when we wrote mails :) 
I thought you meant "make add() private". Sorry for my confusion.